src/protocol/resp/src/parse/parser.rs

```diff
@@ -0,0 +1,434 @@
+// Copyright 2023 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use std::str::FromStr;
+
+use crate::parse::{ParseError, ParseResult};
+#[derive(Clone)]
+pub struct Parser<'a> {
+    data: &'a [u8],
+}
+
+impl<'a> Parser<'a> {
```

Comment:
I'm a Rust noob, so take this with a grain of salt, but do you think naming this lifetime is worthwhile, like `'buffer ` or something? I know simple sequential lifetimes like `'a` are quite common in Rust

Reply:
This ends up being a bit of a judgement call in my experience :)

Personally, if there's only one lifetime and it's used everywhere then I'll use `'a` for it and not worry, others may choose different. If there are multiple lifetimes around then I'll try and give them more useful names. I'll still keep them short, though, since lifetimes tend to be typed out quite a bit.